### DISCUSSION TITLE
How to upgrade Deployed PocketBase with new version ?

### DISCUSSION THREAD

**Discussion by indrazm** [pocketbase/pocketbase#1243] [Q&A]
Hey there! 

I have deployed my project with Pocketbase 0.7.2 at Fly.io.
Is there any way i do upgrade to the latest version ?

Many Thanks!

**Comment by ganigeorgiev** [comment]
PocketBase v0.8+ comes with some breaking changes and requires manual migration steps (including the SDKs).

There are several ways you can upgrade, but I recommend first to apply the changes to your project locally.

1. Download a local backup of your production fly `pb_data`. For example:
   ```sh
   # this will register a ssh key with your local agent (if you haven't already)
   flyctl ssh issue --agent

   # proxies connections to a fly VM through a Wireguard tunnel
   flyctl proxy 10022:22

   # run in a separate terminal to copy the pb_data directory
   scp -r -P 10022 root@localhost:/pb/pb_data  /your/local/pb_data
   ```

2. Download v0.8+ (or v0.9+; both contains the pb_data `upgrade` command) - https://github.com/pocketbase/pocketbase/releases.

3. Follow the "Migrate from v0.7.x" guide in  https://github.com/pocketbase/pocketbase/blob/master/CHANGELOG.md#migrate-from-v07x
   _(only the "Data" and "SDKs" steps are required if you are not using PocketBase as Go framework)_
   
After you've applied the required migration changes to your code and verified that everything is working as expected:

1. modify your docker file (or fly config) to use the latest PocketBase version

2. Create a local backup of your remote pb_data similar to the initial v1) in case something goes wrong.

3. SSH into the fly vm and execute the `./pocketbase upgrade` command.

Alternatively, you can also run the last step locally on top of the downloaded backup (you can set a custom PocketBase `pb_data` with the `--dir` flag) and then reupload it to fly.io.
The reupload process is similar to the backup from 1) but with reversed scp arguments. Note that you may have to use a temporary new name for the `pb_data` (eg. `pb_data_new`) since scp could result in conflict of existing dir/file. Then ssh in your vm (eg. `fly ssh sftp shell`), delete the old pb_data, rename pb_data_new -> pb_data and finally restart your vm._
